pkg/manifests/manifests.go

```diff
@@ -497,6 +474,19 @@ func newExternalDNSDeployment(conf *config.Config, configMapHash string) *appsv1
 			},
 		},
 	}
+	if conf.DNSZonePrivate {
```

Comment:
Can we DRY out the flags a bit?

Something like this:


```go
// Only set shared values in the struct
						Args: []string{
							"--source=ingress",
							"--interval=3m0s",
							"--txt-owner-id=" + conf.DNSRecordID,
							"--domain-filter=" + conf.DNSZoneDomain,
						},

// Then append relevant flags using logic
var additionalArgs []string
if conf.DNSZonePrivate {
		additionalArgs = []string{
			fmt.Sprintf("--provider=%s", azurePrivateDNSProvider),
			fmt.Sprintf("--azure-subscription-id=%s", conf.DNSZoneSub),
		}
} else {
		additionalArgs = []string{
			"--provider=azure",
		}
}
deploymentObj.Spec.Template.Spec.Containers[0].Args = append(deploymentObj.Spec.Template.Spec.Containers[0].Args, additionalArgs...)
```

Reply:
yep that totally makes sense!